jsonnet/github_action.jsonnet

```diff
@@ -1,25 +1,69 @@
+local onMaster = { 'if': "${{ github.ref == 'refs/heads/master' && github.repository == 'jsonnet-libs/k8s' }}" };
+local onPR = { 'if': "${{ github.ref != 'refs/heads/master' && github.repository == 'jsonnet-libs/k8s' }}" };
+
 function(libs) {
-  local jobs = std.split(libs, ','),
   '.github/workflows/main.yml':
-    '# Generated by `make configure`\n' + std.manifestYamlDoc({
+    '# Generated by `make configure`, please do not edit manually.\n' + std.manifestYamlDoc({
       on: [
         'push',
       ],
       jobs: {
-        [std.strReplace(job, '/', '_')]: {
-          name: 'Generate ' + job + ' Jsonnet library and docs',
+        [lib.name]: {
+          name: 'Generate ' + lib.name + ' Jsonnet library and docs',
+          needs: 'terraform',
           'runs-on': 'ubuntu-latest',
           steps: [
-            {
-              uses: 'actions/checkout@v2',
+            { uses: 'actions/checkout@v2' },
+            onMaster {
+              run: |||
+                mkdir ~/.ssh
+                echo "${{ secrets.DEPLOY_KEY }}" > ~/.ssh/id_rsa
+                chmod 600 ~/.ssh/id_rsa
+                export GEN_COMMIT=1
+              |||,
+            },
+            { run: 'make build libs/' + lib.name },
+          ],
+        }
+        for lib in libs
+      } + {
+        what_is_this: {
+          name: 'what is this',
+          'runs-on': 'ubuntu-latest',
+          steps: [
+            { run: 'echo isMaster? ' + onMaster['if'] },
+            { run: 'echo isPR? ' + onPR['if'] },
+            { run: 'echo ${{ github.repository }}' },
+            { run: 'echo ${{ github.ref }}' },
+            { run: 'echo ${{ github.event_name }}' },
+          ],
+        },
+        terraform: {
+          local tf_env = {
+            'working-directory': 'tf',
+            env: {
```

Comment:
To have a nicer output, I'd also set `TF_IN_AUTOMATION` to true: https://www.terraform.io/docs/cli/config/environment-variables.html#tf_in_automation